contracts/test/DualOptionVesting.t.sol

```diff
@@ -0,0 +1,163 @@
+        vm.warp(block.timestamp + 91 days);
+
+        vm.prank(user2);
+        vesting.claimDelayed();
+
+        assertEq(token.balanceOf(user2), 1000 ether);
+    }
+
+    function testChooseOptionBDelayedPartialClaim() public {
+        registerSingle(user3, 1000 ether);
+
+        vm.prank(user3);
+        vesting.chooseOption(false);
+
+        // Wait 30 days (~33% into vesting)
+        vm.warp(block.timestamp + 30 days);
+
+        vm.prank(user3);
+        vesting.claimDelayed();
+
+        // unlocked = 5% + 30/90 * 95% = 5 + 31 = 36% (integer division truncates)
+        uint256 expected = 360 ether; // 36% of 1000 ether
+        assertApproxEqAbs(token.balanceOf(user3), expected, 1e14);
+    }
+
+    function testDoubleRegistrationReverts() public {
+        // Only use the first slot for single registration
+        users[0] = user1;
+        amounts[0] = 1000 ether;
+        // Truncate arrays to length 1 for single registration
+        address[] memory singleUser = new address[](1);
+        uint256[] memory singleAmount = new uint256[](1);
+        singleUser[0] = user1;
+        singleAmount[0] = 1000 ether;
+
+        vesting.registerBeneficiaries(singleUser, singleAmount);
+
+        vm.expectRevert();
+        vesting.registerBeneficiaries(singleUser, singleAmount);
+    }
+
+    function testDoubleOptionReverts() public {
+        registerSingle(user1, 1000 ether);
+
+        vm.prank(user1);
+        vesting.chooseOption(true);
+
+        vm.prank(user1);
+        vm.expectRevert();
+        vesting.chooseOption(true);
+    }
+
+    function testOverclaimReverts() public {
+        registerSingle(user2, 1000 ether);
+
+        vm.prank(user2);
+        vesting.chooseOption(false);
+
+        // Wait full period and claim
+        vm.warp(block.timestamp + 91 days);
+        vm.prank(user2);
+        vesting.claimDelayed();
+
+        // Try to claim again
+        vm.prank(user2);
+        vm.expectRevert();
+        vesting.claimDelayed();
+    }
+
+    function testRescueTokens() public {
+        // Test that owner can rescue tokens from vesting contract
+        uint256 initialBalance = token.balanceOf(owner);
+        uint256 vestingBalance = token.balanceOf(address(vesting));
+        
+        vesting.rescueTokens(owner, 100_000 ether);
+        
+        assertEq(token.balanceOf(owner), initialBalance + 100_000 ether);
+        assertEq(token.balanceOf(address(vesting)), vestingBalance - 100_000 ether);
+    }
+
+    function testRescueTokensRevertsForNonOwner() public {
+        vm.prank(user1);
+        vm.expectRevert();
+        vesting.rescueTokens(user1, 1000 ether);
+    }
+
+    function registerSingle(address user, uint256 amount) internal {
+        address[] memory singleUser = new address[](1);
+        uint256[] memory singleAmount = new uint256[](1);
+        singleUser[0] = user;
+        singleAmount[0] = amount;
+
+        vesting.registerBeneficiaries(singleUser, singleAmount);
```

Comment:
Would be great to confirm the registration works, info is on chain and/or it is claimable